src/lib/create/createTask.ts

```diff
@@ -0,0 +1,66 @@
+import { HttpStatusCode, IModify } from '@rocket.chat/apps-engine/definition/accessors';
+import { IRoom } from '@rocket.chat/apps-engine/definition/rooms';
+import { UIKitViewSubmitInteractionContext } from '@rocket.chat/apps-engine/definition/uikit';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+
+import { TodoistApp } from '../../../TodoistApp';
+import { ModalsEnum } from '../../enums/Modals';
+import { getTasksUrl } from '../../helpers/const';
+
+export async function createTask({
+  app,
+  context,
+  room,
+  modify,
+}: {
+  app: TodoistApp;
+  context: UIKitViewSubmitInteractionContext;
+  room?: IRoom;
+  modify: IModify;
+}) {
+  const data = context.getInteractionData();
+  const state = data.view.state;
+  const user: IUser = context.getInteractionData().user;
+  const project_id = state?.[ModalsEnum.PROJECT_ID_BLOCK]?.[ModalsEnum.PROJECT_ID_INPUT];
+  const taskName = state?.[ModalsEnum.TASK_NAME_BLOCK]?.[ModalsEnum.TASK_NAME_INPUT];
+  const taskPriority =
+    state?.[ModalsEnum.TASK_PRIORITY_BLOCK]?.[ModalsEnum.TASK_PRIORITY_ACTION_ID];
+  const taskDescription =
+    state?.[ModalsEnum.TASK_DESCRIPTION_BLOCK]?.[ModalsEnum.TASK_DESCRIPTION_INPUT];
+  const taskdueDate = Math.floor(
+    new Date(state?.[ModalsEnum.TASK_DUE_DATE_BLOCK]?.[ModalsEnum.TASK_DUE_DATE_INPUT]).getTime() *
+      1
+  );
+
+  const body = {
```

Comment:
you should do some error handling before calling the http server, this way you can avoid some bogus calls